### PR TITLE
First batch of bug fixes found in the CoreClr test suite.

### PR DESCRIPTION
**Pull request by kumpera**
This addresses compiling b28158/test.il and executing exchangetneg.

### CODE REVIEW

**Comment by alexrp** [comment]
There is a test failure in `pinvoke2.exe` on amd64 which doesn't seem to be happening on master.

**Comment by kumpera** [comment]
build

**Comment by kumpera** [comment]
build

File: mono/mini/mini.c [hunk 1]
```diff
@@ -2914,7 +2914,12 @@ create_jit_info (MonoCompile *cfg, MonoMethod *method_to_compile)
 				 */
 				ei->try_start = (guint8*)ei->try_start - cfg->backend->monitor_enter_adjustment;
 			}
-			tblock = cfg->cil_offset_to_bb [ec->try_offset + ec->try_len];
+			if (ec->try_offset + ec->try_len < header->code_size)
+				tblock = cfg->cil_offset_to_bb [ec->try_offset + ec->try_len];
+			else
+				tblock = cfg->bb_exit;
+			if (cfg->verbose_level)
```
Comment (lewurm) [inline mono/mini/mini.c]: how about `if (G_UNLIKELY (cfg->verbose_level >= 4))`? at least that seems to be the verbosity level used around here.

**Comment by kumpera** [comment]
build

**Comment by kumpera** [comment]
This is buggy, DON'T MERGE.

**Pull request merged**